stellar-horizon/debian/stellar-horizon.postinst

```diff
@@ -22,21 +22,31 @@ DBNAME=stellar;
 case "$1" in
   configure)
     # check stellar user exists
-    if ! getent passwd stellar > /dev/null ;then
+    if ! getent passwd stellar > /dev/null; then
       adduser --system --group --quiet --home /var/lib/stellar \
         --no-create-home --disabled-password --shell /bin/bash $ROLE;
     fi
 
     chown -R stellar:stellar /var/lib/stellar/ /var/log/stellar/
 
     # run db migrations
-    if [ -x /usr/bin/stellar-horizon ];then
+    # Confirm the stellar role exists in Postgres before proceeding. If it's not there then no DB migration should proceed
+
+    if [ -x /usr/bin/stellar-horizon ]; then
       if [ -f /etc/default/stellar-horizon ]; then
         # extract --db_url from /etc/default/stellar-horizon
         db_connection_string=$(grep -r '^DATABASE_URL' /etc/default/stellar-horizon | sed 's/DATABASE_URL=//g' | sed 's/"//g');
-        # only apply migrations if history_ledgers table exists
-        if echo "SELECT 'horizon' FROM information_schema.tables WHERE table_schema = 'public' AND table_name = 'history_ledgers';" | runuser -l ${ROLE} -c "psql '${db_connection_string}' -tA" | grep 'horizon' 2>&1 > /dev/null;then
-          sudo -u ${ROLE} stellar-horizon --db-url "$db_connection_string" db migrate up && echo 'info: migrated database'
+        # check `stellar` role and DATABASE_URL
+        if echo "SELECT 'role_exists' FROM pg_roles WHERE rolname='${ROLE}'" | runuser -l ${ROLE} -c "psql '${db_connection_string}' -tA" | grep 'role_exists' 2>&1 > /dev/null; then
```

Comment:
Looks like we need to redirect `stderr` to `stdout` for the `runuser` commands, otherwise we still get a `psql: FATAL:  database "horizon" does not exist` warning displayed when the database doesn't exist or when the table doesn't exist.